web-app/HeimScripts/heatmap/run.r

```diff
@@ -2,16 +2,31 @@ library(jsonlite)
 library(reshape2)
 
 
-main <- function(max_rows=50){
-  df <- loaded_variables[[1]] # SmartR does not support multiple HDD nodes yet
+
+
+main <- function(max_rows=100){
+  df <- loaded_variables[[length(loaded_variables)]] # SmartR does not support multiple HDD nodes yet
+  # We take the last df to alleviate bug with new HDDs dropped being ignored - later on we will use label name explicitly.
   if(ncol(df) > 3){
-    variances <- apply(df[,3:ncol(df)],1,var) # Calculating variance per probe
+  #this is the case for more than 1 sample
+    variances <- apply(df[,3:ncol(df)],1,var, na.rm = T) # Calculating variance per probe (per row)
+    means <- rowMeans(df[,3:ncol(df)], na.rm = T) # this is just an auxiliary column - it will not be used for JSON.
+    sdses <- apply(df[,3:ncol(df)],1,sd, na.rm = T) # this is just an auxiliary column - it will not be used for JSON.
+    df["MEAN"] <- means
+    df["SD"] <- sdses
     df["SIGNIFICANCE"] <- variances
     df <- df[with(df, order(-SIGNIFICANCE)), ]
   }
   else{
-    df["SIGNIFICANCE"] <- rep(1.0, nrow(df))
+  #one sample
+    variance <- 1.0 # We cannot get any significance measure for just one sample
+    mean <- mean(ex_df[,3], na.rm = T) #For just one sample we take mean over all genes (whole column), not per gene
```

Comment:
"#For just one sample we take mean over all genes (whole column), not per gene" This doesn't make sense to me. Why do this to begin with?